### PR TITLE
endgame mac + fixing endgame - massdrop/ctrl 

### PR DESCRIPTION
**Pull request by rshutt**
## Description

This pr both adds an `endgame_mac` keymap and fixes the `endgame` keymap. 

Changes introduced in #12097 (4b96d5876eb01dfd8063949a2e5cb4b70e01786d) actually broke the endgame keymap as the function `md_rgb_matrix_indicators` was replaced with `md_rgb_matrix_indicators_advanced`.  The former had no args and the latter took a `led_min` and `led_max` integer argument.  Turns out, min = 0 and max = ISSI3733_LED_COUNT

Furthermore, with the endgame_mac keymap, I changed some of the matrix effects and set the default to be something I liked :). As expected, the endgame_mac keymap takes into account the swapped cmd and alt keys!

## Types of Changes

- [ ] Core
- [ X ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ X ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Not sure what issues are fixed.  I didn't bother to create an issue

## Checklist

- [ X ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ X ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ X ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ X ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by rshutt** [comment]
> LGTM. Thanks for the fix.

Anytime :). I did it for selfish reasons since I have two of these keyboards.  Maybe I might TAL at what LOE is required to do the things required to save this device from FOSS gulag :/

File: keyboards/massdrop/ctrl/keymaps/endgame_mac/keymap.c [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+                        rgb_matrix_set_flags(LED_FLAG_ALL);
+                        rgb_matrix_enable_noeeprom();
+                    }
+                    break;
+                }
+                return false;
+            // ======================================================== CUSTOM KEYCOADS BELOW ========================================================
+            case COPY_ALL:
+                // Selects all and text and copy
+                SEND_STRING(SS_LCTRL("ac"));
```
[the remark this answers is not in the document]
Comment (rshutt) [reply keyboards/massdrop/ctrl/keymaps/endgame_mac/keymap.c:313]: Is that right?  LCTRL doesn't work?

File: keyboards/massdrop/ctrl/keymaps/endgame_mac/rules.mk [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# RGBLIGHT_ENABLE = no				# Not for MD boards. This is here in case you forget.
+COMMAND_ENABLE = yes				# Commands for debug and configuration
+# BOOTMAGIC_ENABLE = full			# Virtual DIP switch configuration
+# BOOTMAGIC_ENABLE = lite
```
[the remark this answers is not in the document]
Comment (rshutt) [reply keyboards/massdrop/ctrl/keymaps/endgame_mac/rules.mk]: Perhaps I am not sure what you are asking here?  You want me to remove the commented BOOTMAGIC_ENABLE lines for full and lite?

File: keyboards/massdrop/ctrl/keymaps/endgame_mac/keymap.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/massdrop/ctrl/keymaps/endgame_mac/keymap.c:313]: MacOS basically swaps the control and gui functionality. 

What might work better is:
```suggestion
                register_code(keycode_config(KC_LCTL));
                tap_code(KC_A);
                tap_code(KC_C);
                unregister_code(keycode_config(KC_LCTL));
```

The `keycode_config` is from the magic feature, and will swap the code based on the Control-GUI swap option.  That would make it windows and mac friendly. 

Or, simply, just: 
```suggestion
                SEND_STRING(SS_LGUI("ac"));
```

File: keyboards/massdrop/ctrl/keymaps/endgame_mac/config.h [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+// #define RGBLIGHT_EFFECT_CHRISTMAS_INTERVAL 1000     // How long to wait between light changes for the "Christmas" animation, in milliseconds
+// #define RGBLIGHT_EFFECT_CHRISTMAS_STEP 2            // The number of LEDs to group the red/green colors by for the "Christmas" animation
+// #define RGBLIGHT_EFFECT_KNIGHT_LED_NUM RGBLED_NUM   // The number of LEDs to have the "Knight" animation travel
+// #define RGBLIGHT_EFFECT_KNIGHT_LENGTH 3             // The number of LEDs to light up for the "Knight" animation
+// #define RGBLIGHT_EFFECT_KNIGHT_OFFSET 0             // The number of LEDs to start the "Knight" animation from the start of the strip by
+// #define RGBLIGHT_RAINBOW_SWIRL_RANGE 255            // Range adjustment for the rainbow swirl effect to get different swirls
+// #define RGBLIGHT_EFFECT_SNAKE_LENGTH 4              // The number of LEDs to light up for the "Snake" animation
+
+// This list in in the correct mode order. Next mode is the following line, previous mode is previous line. Loops around.
+#define DISABLE_RGB_MATRIX_SOLID_COLOR                   // Static single hue, no speed support
```
Comment (drashna) [inline keyboards/massdrop/ctrl/keymaps/endgame_mac/config.h:81]: these are no longer used.

File: keyboards/massdrop/ctrl/keymaps/endgame_mac/rules.mk [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# RGBLIGHT_ENABLE = no				# Not for MD boards. This is here in case you forget.
+COMMAND_ENABLE = yes				# Commands for debug and configuration
+# BOOTMAGIC_ENABLE = full			# Virtual DIP switch configuration
+BOOTMAGIC_ENABLE = lite
```
Comment (drashna) [inline keyboards/massdrop/ctrl/keymaps/endgame_mac/rules.mk:4]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/massdrop/ctrl/keymaps/endgame_mac/config.h (hunk 3 above)
Comment (rshutt-va) [reply keyboards/massdrop/ctrl/keymaps/endgame_mac/config.h:81]: When you say "these" are you including the other #define DISABLE.*'s seen listed afterwards?  

e.g. DISABLE_RGB_MATRIX_ALPHAS_MOD... DISABLE_RGB_MATRIX_GRADIENT_UP_DOWN... etc?

File: keyboards/massdrop/ctrl/keymaps/endgame_mac/rules.mk (hunk 4 above)
Comment (rshutt-va) [reply keyboards/massdrop/ctrl/keymaps/endgame_mac/rules.mk]: Fixed this one.

File: keyboards/massdrop/ctrl/keymaps/endgame_mac/config.h (hunk 3 above)
Comment (drashna) [reply keyboards/massdrop/ctrl/keymaps/endgame_mac/config.h:81]: the `disable` defines are no used anymore, instead you have to explicitly enable the animation modes

Comment (rshutt-va) [reply keyboards/massdrop/ctrl/keymaps/endgame_mac/config.h:81]: So if I nuke all of them, is the PR ok to merge?

File: keyboards/massdrop/ctrl/keymaps/endgame_mac/keymap.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (rshutt) [reply keyboards/massdrop/ctrl/keymaps/endgame_mac/keymap.c:313]: Ok so what I know is that I simply edited an existing keymap to get this one kicking...  And I'm currently using it!  So whatever doesn't work isn't something I've needed.  Ill take a look, but at this point, the juice is no longer worth the squeeze.  It works for me, I put in a PR to try to make it work for others, but apparently, that ain't enough.

File: keyboards/massdrop/ctrl/keymaps/endgame_mac/config.h (hunk 3 above)
Comment (rshutt) [reply keyboards/massdrop/ctrl/keymaps/endgame_mac/config.h:81]: @drashna I'm trying to understand if I make the changes you are suggesting in this comment, will my PR be approved and merged?  Or are there yet other changes that have to be made?  The level of effort here for making this PR part of the code base is very quickly eclipsing any personal value I would receive from the same.

Comment (drashna) [reply keyboards/massdrop/ctrl/keymaps/endgame_mac/config.h:81]: We switched from an implicitly enabled unless explicitly disabled approach, to an explicitly enabled approach.

So the "disable mode" defines simply don't do anything anymore and should be removed. 

Instead, you want to use the "enable mode" defines for the modes that you want. 

This should be the full list of animation modes available:
```c
#    define ENABLE_RGB_MATRIX_ALPHAS_MODS
#    define ENABLE_RGB_MATRIX_GRADIENT_UP_DOWN
#    define ENABLE_RGB_MATRIX_GRADIENT_LEFT_RIGHT
#    define ENABLE_RGB_MATRIX_BREATHING
#    define ENABLE_RGB_MATRIX_BAND_SAT
#    define ENABLE_RGB_MATRIX_BAND_VAL
#    define ENABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
#    define ENABLE_RGB_MATRIX_BAND_PINWHEEL_VAL
#    define ENABLE_RGB_MATRIX_BAND_SPIRAL_SAT
#    define ENABLE_RGB_MATRIX_BAND_SPIRAL_VAL
#    define ENABLE_RGB_MATRIX_CYCLE_ALL
#    define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
#    define ENABLE_RGB_MATRIX_CYCLE_UP_DOWN
#    define ENABLE_RGB_MATRIX_RAINBOW_MOVING_CHEVRON
#    define ENABLE_RGB_MATRIX_CYCLE_OUT_IN
#    define ENABLE_RGB_MATRIX_CYCLE_OUT_IN_DUAL
#    define ENABLE_RGB_MATRIX_CYCLE_PINWHEEL
#    define ENABLE_RGB_MATRIX_CYCLE_SPIRAL
#    define ENABLE_RGB_MATRIX_DUAL_BEACON
#    define ENABLE_RGB_MATRIX_RAINBOW_BEACON
#    define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS
#    define ENABLE_RGB_MATRIX_RAINDROPS
#    define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
#    define ENABLE_RGB_MATRIX_HUE_BREATHING
#    define ENABLE_RGB_MATRIX_HUE_PENDULUM
#    define ENABLE_RGB_MATRIX_HUE_WAVE
#    define ENABLE_RGB_MATRIX_PIXEL_RAIN
#    define ENABLE_RGB_MATRIX_PIXEL_FLOW
#    define ENABLE_RGB_MATRIX_PIXEL_FRACTAL
// enabled only if RGB_MATRIX_FRAMEBUFFER_EFFECTS is defined
#    define ENABLE_RGB_MATRIX_TYPING_HEATMAP
#    define ENABLE_RGB_MATRIX_DIGITAL_RAIN
// enabled only of RGB_MATRIX_KEYPRESSES or RGB_MATRIX_KEYRELEASES is defined
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
#    define ENABLE_RGB_MATRIX_SPLASH
#    define ENABLE_RGB_MATRIX_MULTISPLASH
#    define ENABLE_RGB_MATRIX_SOLID_SPLASH
#    define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
```

**Pull request closed without merging**